nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,430 @@
+                                ' interpolation')
+    target = File(exists=True, argstr="-target %s", xor=['transform'],
+                  desc='output volume specification read from the target '
+                  'volume if specified')
+    translation = traits.Tuple((traits.Float(), traits.Float(),
+                                traits.Float()),
+                               desc='translation (x,y,z) in mm',
+                               argstr='-translation %g %g %g',
+                               xor=['transform'])
+    euler = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
+                         desc='(theta, phi, psi) in degrees',
+                         xor=['transform'], argstr='-euler %g %g %g')
+    deformation = traits.Tuple((traits.Float(),) * 6,
+                               desc='(xx,yy,zz,xy,yz,xz)', xor=['transform'],
+                               argstr='-deformation %g %g %g %g %g %g')
+
+
+class AffScalarVolOutputSpec(TraitedSpec):
+    out_file = File(desc='moved volume', exists=True)
+
+
+class AffScalarVol(CommandLineDtitk):
     """
     Applies affine transform to a scalar volume
 
-                Example
-                -------
-
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.affScalarVolTask()
-                >>> node.inputs.in_volume = 'fa.nii.gz'
-                >>> node.inputs.in_xfm = 'ants_Affine.txt'
-                >>> node.run() # doctest: +SKIP
-                """
-    input_spec = affScalarVolInputSpec
-    output_spec = affScalarVolOutputSpec
+    Example
+    -------
+
+    >>> from nipype.interfaces import dtitk
+    >>> node = dtitk.AffScalarVol()
+    >>> node.inputs.in_file = 'im1.nii'
+    >>> node.inputs.transform = 'im_affine.aff'
+    >>> node.cmdline
+    'affineScalarVolume -in im1.nii -interp 0 -out im1_affxfmd.nii -trans
+     im_affine.aff'
+    >>> node.run() # doctest: +SKIP
+    """
+    input_spec = AffScalarVolInputSpec
+    output_spec = AffScalarVolOutputSpec
     _cmd = 'affineScalarVolume'
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = os.path.abspath(self.inputs.out_file)
-        return outputs
+    def _format_arg(self, name, spec, value):
+        if name == 'interpolation':
+            value = {'trilinear': 0, 'NN': 1}[value]
+        return super(AffScalarVol, self)._format_arg(name, spec, value)
+
+
+class DiffeoSymTensor3DVolInputSpec(CommandLineInputSpec):
+    in_file = File(desc='moving tensor volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = File(desc='output filename',
+                    argstr="-out %s", name_source="in_file",
+                    name_template="%s_diffeoxfmd", keep_extension=True)
+    transform = File(exists=True, argstr="-trans %s",
+                     mandatory=True, desc='transform to apply')
+    df = traits.Str('FD', argstr="-df %s", usedefault=True)
```

Comment:
Description?

And if `-df` were omitted, it would act as if `-df FD` were passed? Are there no other options?

Reply:
Here is the documentation. I think that is the case... "df" usually means "deformation field" in this package, but I'm not sure what FD stands for (I didn't find it in the documentation).

Reply:
Gotcha. Okay, this makes sense. It sets the default, but because it's `Str` not `Enum`, it lets users who know more than we do set an alternative.